### ISSUE TITLE
[03-13] YOLO#2

### ISSUE THREAD

**Issue by mingnuj** [dev-writeup-2024/march#32]
어제에 이은 YOLO 입니다. 여담이지만 You Live Only Once는 한번 사는 인생이니 즐겁게 살아라! 이게 아니고 최선을 다해 살라고 해석하고 삽니다. 다들 열심히 살아 보자구요. 그치만 노력의 기준은 상대적인 것이니 무리하진 않았으면 합니다.

* [[03-12] YOLO#1: v1, v2](https://github.com/dev-writeup-2024/march/issues/29)

# YOLO v3

YOLO v1은 2016년에 나왔고, YOLO v2는 2017년에 나왔다.  이 작자는 2018년에 YOLO v3를 낸다. 이번엔 자기 맘대로 편하게 써서 아카이브에 올렸다. 이상한 사람이다.

![image](https://github.com/dev-writeup-2024/march/assets/25028093/1684e672-6463-4fba-917f-bb6517baadd3)
단적으로 소제목이 이 모양이다. 다른 사람들한테서 좋은 아이디어 끌고 왔단다. 그냥 연구가 신나는 사람인것 같다. 매력적이야..

![image](https://github.com/dev-writeup-2024/march/assets/25028093/1b98da08-d669-4df6-9ad3-0498f80cec3c)
그래프 잘못 그린 줄 알았는데 아주 정상이었다. 그냥 자기들이 개쩐다는걸 이렇게 보여준 것 같다.

## 논문 정보

- 제목: Yolov3: An incremental improvement
- 저자: Redmon, J., & Farhadi, A. (2018)
- 저널: arXiv preprint arXiv:1804.02767.

놀랍게도 아카이브에 올렸는데 인용 수가 26251회. ㅋㅋ

## 특징

YOLO v2와 크게 변하지 않았다. 다만, 당시의 최신 기법을 사용하여 성능을 높였다. 

### Bounding Box Predication

같은 방법 사용

### Class Predication

* Softmax 대신 binary cross-entropy loss 사용
* 더 복잡한 데이터셋에 강한 모델을 만들 수 있다.

###  Prediction Across Scales

![image](https://github.com/dev-writeup-2024/march/assets/25028093/9a5ba9d1-d0e7-46f5-b3bc-3354048705c5)

* Darknet-53의 최종 feature map에 여러 convolutional layer를 추가해서 upsampling을 2번
  * feature pyramid를 생성하기 위한 용도로 사용한다.
  * 3개의 scale을 사용
  * 최종 feature map과 이전 2개의 feature map은 upsampled 한 feature map과 통합 
  * FPN(Feature Pyramid Network)과 유사한 방식으로 더 풍부한 정보를 활용한다.

### Feature Extractor: DarkNet-53

53개의 conv net을 가지고, ResNet에서 제안한 shor-cut connection을 사용한다.

![image](https://github.com/dev-writeup-2024/march/assets/25028093/8209d4ca-8dd0-466e-8baf-55e7127a9ead)

* ImageNet dataset에서의 Darknet-53의 성능
  ![image](https://github.com/dev-writeup-2024/march/assets/25028093/f1bc911e-5b92-46b0-8d4e-5af10301ba5e)

## YOLO v3 이후 Joseph Redmon...

![image](https://github.com/dev-writeup-2024/march/assets/25028093/33c55f75-f210-4fb2-972a-124a7329b4b4)

신나서 적다가 슬퍼졌다. 본인이 만들어낸 모델이 사회에 미친 영향 때문에 CV 연구를 그만두었다.
기술이란게 악용하는 인간들이 생길 수 밖에 없으니 책임을 느끼는 마음은 이해가 가지만 너무 아쉬운 일이다.

YOLO v4 부터는 다른 사람들이 이어서 만들어내고 있다. 
내가 알고 있는 YOLO도 v3 까지 이므로 내일부터는 아예 새로운 공부 하는 기분으로 적을 것 같다.

# 참고

* [Yolov3: An incremental improvement](https://arxiv.org/abs/1804.02767)
* 티스토리: [[논문 읽기] YOLOv3(2018) 리뷰](https://deep-learning-study.tistory.com/509)
* Medium: [YOLO Creator Joseph Redmon Stopped CV Research Due to Ethical Concerns](https://medium.com/syncedreview/yolo-creator-says-he-stopped-cv-research-due-to-ethical-concerns-b55a291ebb29)
